### PR TITLE
Changes in yarn api client for Enterprise gateway yarn checks.

### PR DESCRIPTION
**Pull request by IMAM9AIS**
Following functions are added:-

1. To access maximum_container_allocation data for the given yarn cluster.
2. To access queue details and partition details.
3. To understand partition availability given memory requirements and threshold limits.

### CODE REVIEW

**Comment by IMAM9AIS** [comment]
@kevin-bates I have created this PR for the changes need in yarn api client for this PR:- 
https://github.com/jupyter/enterprise_gateway/pull/699

File: yarn_api_client/resource_manager.py [hunk 1]
```diff
@@ -421,3 +422,72 @@ def cluster_change_application_priority(self, application_id, priority):
...
+
+    def cluster_scheduler_queue_availability(self, candidate_partition, availabilty_threshold):
+        """
+        Checks whether the requested memory satisfies the available space of the queue
+        This solution takes into consideration the node label concept in cluster.
+        Following node labelling, the resources can be available in various partition.
+        Give the partition data it tells you if the used capacity of this parition is spilling
+        the threshold specified.
+
+        :param cadidate_parition:
```
Comment (kevin-bates) [inline yarn_api_client/resource_manager.py]: Typo in variable name.  We should probably add data types for these doc items - particularly for the non-string types.

File: yarn_api_client/resource_manager.py [hunk 2]
```diff
@@ -421,3 +422,72 @@ def cluster_change_application_priority(self, application_id, priority):
...
+
+    def cluster_scheduler_queue(self, yarn_queue_name):
+        """
+        Given a queue name, this function tries to locate the given queue in the object
+        returned by scheduler endpoint.
+
+        The queue can be present inside a multilevel structure. This solution tries
+        to locate the queue using breadth-first-search algorithm.
+
+        :param yarn_queue_name:
```
Comment (kevin-bates) [inline yarn_api_client/resource_manager.py]: We may want to state that `yarn_queue_name` is case-sensitive.

File: yarn_api_client/resource_manager.py [hunk 3]
```diff
@@ -421,3 +422,72 @@ def cluster_change_application_priority(self, application_id, priority):
...
+        return True
+
+
+    def cluster_queue_partition(self, candidate_queue, cluster_node_label):
+        """
+        A queue can be divided into multiple partitions having different node labels.
+        Given the candidate queue and parition node label, this extracts the partition
+        we are interested in
+        :param candidate_queue:
+        :param cluster_node_label:
```
Comment (kevin-bates) [inline yarn_api_client/resource_manager.py]: Datatype for `candidate_queue` would be good here.  We might want to mention that `cluster_node_label` is case-sensitive.

File: yarn_api_client/resource_manager.py [hunk 4]
```diff
@@ -421,3 +422,72 @@ def cluster_change_application_priority(self, application_id, priority):
...
+        :return: partition object
+        """
+        for partition in candidate_queue['capacities']['queueCapacitiesByPartition']:
+            if partition['partitionName'] == cluster_node_label:
+                return partition
+        return None
```
Comment (kevin-bates) [inline yarn_api_client/resource_manager.py]: Some lint tools (e.g., flake8) require a blank line at EOF.

File: yarn_api_client/resource_manager.py [hunk 5]
```diff
@@ -421,3 +422,72 @@ def cluster_change_application_priority(self, application_id, priority):
...
+    def cluster_scheduler_queue(self, yarn_queue_name):
+        """
+        Given a queue name, this function tries to locate the given queue in the object
+        returned by scheduler endpoint.
+
+        The queue can be present inside a multilevel structure. This solution tries
+        to locate the queue using breadth-first-search algorithm.
+
+        :param yarn_queue_name:
+        :return: queue
```
Comment (kevin-bates) [inline yarn_api_client/resource_manager.py]: Should probably add something like _or `None` if not found_ and perhaps the expected datatype.

File: yarn_api_client/resource_manager.py [hunk 6]
```diff
@@ -421,3 +422,72 @@ def cluster_change_application_priority(self, application_id, priority):
...
+
+
+    def cluster_queue_partition(self, candidate_queue, cluster_node_label):
+        """
+        A queue can be divided into multiple partitions having different node labels.
+        Given the candidate queue and parition node label, this extracts the partition
+        we are interested in
+        :param candidate_queue:
+        :param cluster_node_label:
+        :return: partition object
```
Comment (kevin-bates) [inline yarn_api_client/resource_manager.py]: Should probably add something like _or `None` if not found_ and perhaps the expected datatype.

**Comment by IMAM9AIS** [comment]
@kevin-bates I have made the changes. 
I have also made the changes on this PR :- jupyter/enterprise_gateway#699

**Comment by kevin-bates** [comment]
I'd like to include this in 0.3.6 and will begin that process.

**Pull request merged**